feat: allow hex color codes with leading hash (#)

Monty opened the pull request:
Allow # prefix to color codes i.e. `alt_red: '#ed4a46'`  works just the same as  `alt_red: 'ed4a46'`

I also added a test `fn from_hex_str_with_hash()` to ensure this works as expected

trueNAHO commented on src/color.rs:
```diff
@@ -151,6 +153,15 @@ mod tests {
+
+    let color_short = Color::from_hex_str("#4ec").unwrap();
+    assert_eq!(Color::Rgb(0x44, 0xee, 0xcc), color_short);
+}
```
This should be indented. Consider running `cargo fmt`.

Monty commented:
I'm not sure which vim plugin colorizes #RRGGBB values. I don't think colorizing _all_ 6 digit numbers would be ideal though.

I can work around it easily by adding comments to my theme's colors. Feel free to ignore it.

`red:  "ed4a46"  #ed4a46`

Monty commented:
Your suggestion provided a unique visual perception experiment.

I looked in my .vim/bundle and saw I was using [vim-css-color](https://github.com/ap/vim-css-color/blob/master/README.md), so I compared [colorhighlighter](https://github.com/SilentGlasses/colorhighlighter)

colorhighlighter has more capabilities and features, i.e.:
- RGB/RGBA colors: rgb(255, 0, 0), rgba(255, 0, 0, 0.5)
- HSL/HSLA colors: hsl(0, 100%, 50%), hsla(0, 100%, 50%, 0.5)

However look at these two examples, especially the #368aeb color. Both render the same shade of #368aeb blue according to Digital Color Meter, but the white vs dark backgrounds make them appear quite different. I'll stick with vim-css-color as it's easier for me to mirror the shade I see using ls.

<img width="355" height="182" alt="colorhighlighter" src="https://github.com/user-attachments/assets/3207aa0a-06c3-4d23-8ce6-5283532b4d00" />

<img width="360" height="178" alt="vim-css-color" src="https://github.com/user-attachments/assets/09cb30e6-4348-46b8-b236-09b72c2b9ccd" />

As expected, neither renders hex colors not preceded by a #.

I had deleted my feature branch, so I'm, going to recreate it. 

- It's only a one line change in color.rs
- I ran cargo fmt this time to clean up the test I added
- It doesn't change any functionality in from_hex_str except to remove a leading # if one exists

Feel free to ignore it. I'm fine with recompiling my version as you make changes. But if anyone ever requests this feature it will save you the work.

Let me know if the (false) color differences are as apparent to you as they were to me. I'm not a graphics person so I found it surprising.

Monty commented:
Just a brief note of appreciation for fd, bat, and hyperfine, especially fd for making it simpler to manage cron job created files, e.g.:
`fd -I -t f -d 2 --older 90days "[0-9]{6}.csv|[0-9]{6}.txt" -x rm`
Thanks to you also for uv. I'm now well on my way to replacing pip altogether. 
I've started playing with pastel to help me better understand colors.

Since you like automating things, I'll share one of my timesavers:
[getRustAppVersions.sh](https://github.com/user-attachments/files/25021479/getRustAppVersions.sh)

Monty commented:
Love that! I'll send this to a Luddite friend who is still clinging to his Apple Watch 3.

`uvx norwegianblue apple-watch`
